### PR TITLE
LoginConfig: remove refs to all but the MP-JWT method and make MP-JWT the default one

### PR DESCRIPTION
**Pull request by sberyozkin**
Related #182 

This PR is the first step in the long (possibly never ending) process of dropping `LoginConfig`:
- removes the references to all but `MP-JWT` methods - `MP-JWT` spec should really be only about JWT; note dropping these references from the doc only serves this purpose, i.e, stop sending the message that `LoginConfig` can be an annotation for all sort of auth mechanisms
- `MP-JWT` is a default method now - it does not break the backward compatibility since until now setting the auth method is required, I suspect in the vast majority of cases where this annotation is still used it is `@LoginConfig(authMethod="MP-JWT")` so at least the users will now be able to do just `@LoginConfig`...
- updated the baseline version in this package to `1.1` - I'm scared of this update the most :-), @Emily-Jiang, please check if it is correct

I really don't think we can do more about `@LoginConfig` in the short/medium term
Hi Romain, @rmannibucau, not sure why but could not add you as a reviewer, I'm doing this PR given your last comment at #182 
Also see @teddyjtorres

### CODE REVIEW

**Comment by rmannibucau** [comment]
@sberyozkin think this change does not need to go with a 1.1 change but ok for me as a move in the right direction.

Side question related to this issue plannification: when is mp-jwt-auth 2 planned? Cause makes long enough without any change to drop the annotation and do a 2.0 now (most MP spec broke often so guess jwt auth can too now ;)).

**Comment by sberyozkin** [comment]
Hi @rmannibucau for MP JWT it would be `1.2` which will go into MP 4.0, the process is slow right now so i can only guess MP JWT 2.0 with the breaking changes if any would be in 2021. MP JWT 1.2 has a good number of various non-breaking updates and I could not see/find the scope for going for more and doing a proper 2.0.

**Comment by rmannibucau** [comment]
@sberyozkin ok, just thought that cause of all the pain other specs did with unexpected breaking changes this one could go but also agree it is just a dead annotation so while doc is clear it is already a very good move. Thanks Sergey.

**Review by rmannibucau — APPROVED** [review]

**Comment by Emily-Jiang** [comment]
@sberyozkin don't need to be scared with the version changes. The bnd plugin will work it out for you. When you do `mvn verify` and it builds cleanly, you are good. Otherwise, the plugin will tell you what to do. Your changes look good to me.

**Review by Emily-Jiang — APPROVED** [review]

File: api/src/main/java/org/eclipse/microprofile/auth/LoginConfig.java [hunk 1]
```diff
@@ -37,27 +37,19 @@
 @Retention(RetentionPolicy.RUNTIME)
 public @interface LoginConfig {
     /**
-     * Soften the requirements here:
-     *
      * The authMethod is used to configure the authentication mechanism for the
      * JAX-RS application. As a prerequisite to gaining access to any web resources
      * which are protected by an authorization constraint, a user must have
-     * authenticated using the configured mechanism. Supported values include
-     * "BASIC", "DIGEST", "FORM", "CLIENT-CERT", "MP-JWT", or a vendor-specific
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/auth/LoginConfig.java:46]: Why did you delete these?

Comment (sberyozkin) [reply api/src/main/java/org/eclipse/microprofile/auth/LoginConfig.java:46]: @Emily-Jiang The idea is stop start preparing for the eventual removal of this annotation; there was a long discussion in of the old issues with many people preferring it; but it can also be a very sensitve thing to do in some cases, so what we agreed with Romain is to start with dropping any references to all but `MP-JWT` authentication mechanism for the users not to keep thinking that the MP JWT API introduces the annotation which can be used even for non-MP-JWT flows (which was one of the problems discussed in the earlier discussion), the scope of the MP JWT should be all about MP JWT.

**Comment by sberyozkin** [comment]
Hi @Emily-Jiang thanks, yes, it failed during the local build and suggested `1.0.1` but I thought aligning it with `1.1` would be better.
Having said it, I've been thinking about the whole default `authMethod` enhancement and thought I made a pointless enhancement really as `LoginConfig` on its own does not give any indication at all of what this `LoginConfig` is about and in a way it contradicts with our plan to deprecate/remove this annotation as it makes it easier to use it for the `MP-JWT` auth mechanism.
So I'm going to revert the code/package changes and only trim `LoginConfig` docs for `MP-JWT` only be referenced

**Comment by sberyozkin** [comment]
Hi All, so now it is only the docs update keeping a reference to only `MP-JWT` and sending a message with `Support for the other authentication mechanisms is optional.` that it is really not the way to go with using this `MP-JWT` annotation for all the auth methods out there. The `optional` bit is not breaking I think as TCK does not enforce this annotation is effective anyway. If someone is concerned about `optional` then I will change
`Support for the other authentication mechanisms is optional.` to for example `Other authentication mechanisms may be supported` which is the same really.
But in any case removing the refs to `FORM`/etc cleans it up and will make it easier for us to drop this annotation later on...

**Comment by rdebusscher** [comment]
Should we not mention already that it is the idea to remove the annotation all together later?  It will trigger maybe people to give some scenarios which we are not thinking about today but are important.

File: api/src/main/java/org/eclipse/microprofile/auth/LoginConfig.java (hunk 1 above)
Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/auth/LoginConfig.java:46]: ok, thanks

**Review by radcortez — APPROVED** [review]

**Comment by sberyozkin** [comment]
@rdebusscher Hi, yes, makes sense, will update with the Java docs with something like "Note: this annotation may be removed in the future versions of the MP JWT specification". I suppose adding `@Deprecated` would also be the right thing to do, but with `@Deprecated` it would be good to have an alternative link so I'll probably avoid it.
By the way, is there some similar annotation in Jakarta Security ? I've looked around but could not see anything...Also CC @teddyjtorres

**Comment by sberyozkin** [comment]
@radcortez thanks, I'll do another small update to note the annotation may be removed, and then it should be ready for a merge

**Comment by sberyozkin** [comment]
@rdebusscher So I've added a note that this annotation can be removed which will let some users ask the questions if they need to before it is probably gone in 2.0/etc.

**Comment by sberyozkin** [comment]
Going with merge now, any concerns - please open a new issue
